content/blog/2021/06/writing-your-first-policy-with-kubewarden.md

```diff
@@ -0,0 +1,554 @@
+in our [Policy Hub](https://hub.kubewarden.io), chances are that you
+want to check if the policy behaves as you expect -- leaving aside the
+fact that you might want to audit it; we will cover it in a future
+blog post. --
+
+Enough of an introduction. Let's get our feet wet!
+
+## The policy idea
+
+Let's write a simple policy that has to do with internal company
+compliance. Let's say that in your organization, all teams have an
+identifier, and that all the namespaces in your cluster must meet the
+following requirements:
+
+- Name is prefixed with the team identifier owning the namespace
+    - Due to internal company compliance, all team identifiers follow
+      the pattern `t-\w+-\w+`.
+- The following annotation keys are required. The annotation names are
+  also part of the compliance rules and agreed upon within the
+  company:
+    - `compliance.my-company.com/team-contact-email`
+        - Must be a syntactically valid email ending with our company
+          domain.
+    - `compliance.my-company.com/team-region`
+        - Allowed values: `AMER` (North, Central and South America),
+          `APAC` (Asia-Pacific), `EMEA` (Europe, Middle-East and
+          Africa).
+
+The policy won't allow the creation of new namespaces that don't
+follow any of this rules.
+
+That's it for now. I am sure you already have in mind several ways to
+enrichen, improve and secure this simple policy.
+
+## The language
+
+At the time of writing we have SDKs for the following languages:
+
+- [Rust](https://github.com/kubewarden/policy-sdk-rust)
+- [Swift](https://github.com/kubewarden/policy-sdk-swift)
+- [TinyGo](https://github.com/kubewarden/policy-sdk-go)
+
+This list is not static and more will be implemented over time. If you
+want to know more about the SDKs and the languages we support and why,
+you can [extend the read through the Kubewarden
+book](https://docs.kubewarden.io/writing-policies/index.html#programming-language-requirements).
+
+We are going to write our policy in Go, thus, we will use the TinyGo
+compiler.
+
+In general, starting a new policy is a matter of using the policy
+template we have on GitHub, except for Rust, that has a dedicated
+tool. You can find more information [in the
+book](https://docs.kubewarden.io/).
+
+## The policy
+
+We head to the [`go-policy-template`
+repo](https://github.com/kubewarden/go-policy-template) and use that
+template, creating a repository in one of our GitHub organizations.
+
+### Prerequisites
+
+We clone the repository we just generated from the template locally.
+
+> **Note**: you will need `docker` or `podman` installed for this
+> exercise. At the time of writing, the policy template's `Makefile`
+> calls to the `docker` CLI, so if you use `podman`, make sure you
+> also have a `docker` wrapper or symlink in your `$PATH`. This is so
+> you don't have to install TinyGo yourself.
+
+Let's perform a sanity check by running `make`:
+
+```console
+$ rm -rf vendor
```

Comment:
That should be a `make clean` ;-)

Reply:
`make clean` already exists, and is a target that removes the `policy.wasm` binary artifact. Since removing the `vendor` folder is just a way to avoid extra commands when following the blog post (such as `go mod tidy` and `go mod vendor`), making the blog post shorter and less cumbersome, I think it makes sense to just remove it as a first step. But is something that someone writing a real policy wouldn't really do, they would just `go mod tidy` and `go mod vendor`, or not vendor at all, depends on what each developer prefers.

Reply:
I would leave `rm -rf` as it is